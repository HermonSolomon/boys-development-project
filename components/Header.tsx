import { Box, Flex } from "@chakra-ui/react";
import React from "react";
import Wrapper from "./Wrapper";

const Header = () => {
  return (
    <Wrapper>
      <Flex>Box</Flex>
    </Wrapper>
  );
};

export default Header;
