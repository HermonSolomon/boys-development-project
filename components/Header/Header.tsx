import React, { useState } from "react";
import {
  Box,
  Flex,
  Text,
  Button,
  Stack,
  IconButton,
  useDisclosure,
  useColorModeValue,
} from "@chakra-ui/react";
import { HamburgerIcon, CloseIcon } from "@chakra-ui/icons";
import Link from "next/link";
export const Header = () => {
  const { isOpen, onToggle } = useDisclosure();

  interface NavLink {
    link: string;
    path: string;
  }

  const navLinks: NavLink[] = [
    {
      link: "Home",
      path: "/",
    },
    {
      link: "About",
      path: "/about",
    },
    {
      link: "Programme",
      path: "/programme",
    },
    {
      link: "Blog",
      path: "/blog",
    },
    {
      link: "Resources",
      path: "/resources",
    },
    {
      link: "Contact",
      path: "/contact",
    },
  ];

  return (
    <Box>
      <Flex
        as="nav"
        align="center"
        justify="space-between"
        wrap="wrap"
        padding={{ base: 2, md: 4 }}
        bg={useColorModeValue("gray.100", "gray.900")}
        // height={{ base: "100vh", sm: "auto", md: "auto" }}
        height="calc(var(--menu-height) -var(--wix-ads-height))"
        color={useColorModeValue("gray.800", "white")}
      >
        <Box display={{ base: "block", md: "none" }}>
          <IconButton
            onClick={onToggle}
            icon={isOpen ? <CloseIcon /> : <HamburgerIcon />}
            variant="ghost"
            aria-label="Toggle Navigation"
          />
        </Box>
        <Flex
          align="center"
          justify={{ base: "center", md: "space-between" }}
          w="100%"
          py={{ base: 2, md: 0 }}
        >
          <Text fontSize="lg" fontWeight="bold">
            Logo
          </Text>
          <Box
            display={{ base: isOpen ? "block" : "none", md: "flex" }}
            flexBasis={{ base: "100%", md: "auto" }}
            flexGrow={1}
            alignItems="center"
            justifyContent="center"
            mt={{ base: 4, md: 0 }}
            borderTop={{ base: "1px", md: "none" }}
          >
            <Stack
              spacing={4}
              align="center"
              justify={["center", "space-between", "flex-end", "flex-end"]}
              direction={["column", "row", "row", "row"]}
              pt={[4, 4, 0, 0]}
            >
              {navLinks.map((link: NavLink) => (
                <Link href={{ pathname: link.path }} key={link.path}>
                  {link.link}
                </Link>
              ))}
            </Stack>
          </Box>
        </Flex>
      </Flex>
    </Box>
  );
};
