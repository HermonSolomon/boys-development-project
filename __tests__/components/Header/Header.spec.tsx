import React from "react";
import { render, fireEvent, getByText, waitFor } from "@testing-library/react";
import { Header } from "@/components/Header/Header";
import "@testing-library/jest-dom";

describe("Header component", () => {
  const navLinks = [
    "Home",
    "About",
    "Programme",
    "Resources",
    "Blog",
    "Contact",
  ];

  it("renders the header logo", () => {
    const { getByText } = render(<Header />);
    const logo = getByText("Logo");
    expect(logo).toBeInTheDocument();
  });

  it("opens and closes the navigation menu when hamburger icon is clicked", () => {
    const { getByLabelText, getByText, queryByText } = render(<Header />);
    const hamburgerIcon = getByLabelText("Toggle Navigation");
    fireEvent.click(hamburgerIcon);
    navLinks.forEach((link) => {
      expect(getByText(link)).toBeInTheDocument();
    });
    fireEvent.click(hamburgerIcon);
    navLinks.forEach(async (link) => {
      await waitFor(() => {
        expect(queryByText(link)).not.toBeInTheDocument();
      });
    });
  });

  it("renders all navigation links", () => {
    const { getByText } = render(<Header />);
    navLinks.forEach((link) => {
      expect(getByText(link)).toBeInTheDocument();
    });
  });
});
