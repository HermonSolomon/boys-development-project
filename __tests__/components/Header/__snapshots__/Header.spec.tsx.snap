// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component matches the snapshot 1`] = `
<div>
  <div
    class="css-0"
  >
    <nav
      class="css-17md98w"
    >
      <div
        class="css-dsyim5"
      >
        <button
          aria-label="Toggle Navigation"
          class="chakra-button css-p7mkev"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="chakra-icon css-onkibi"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M 3 5 A 1.0001 1.0001 0 1 0 3 7 L 21 7 A 1.0001 1.0001 0 1 0 21 5 L 3 5 z M 3 11 A 1.0001 1.0001 0 1 0 3 13 L 21 13 A 1.0001 1.0001 0 1 0 21 11 L 3 11 z M 3 17 A 1.0001 1.0001 0 1 0 3 19 L 21 19 A 1.0001 1.0001 0 1 0 21 17 L 3 17 z"
              fill="currentColor"
            />
          </svg>
        </button>
      </div>
      <div
        class="css-dnzk3y"
      >
        <p
          class="chakra-text css-1hgmzub"
        >
          Logo
        </p>
        <div
          class="css-51qgs1"
        >
          <div
            class="chakra-stack css-1wzma5t"
          >
            <button
              class="chakra-button css-1k6w77y"
              type="button"
            >
              Home
            </button>
            <button
              class="chakra-button css-1k6w77y"
              type="button"
            >
              About
            </button>
            <button
              class="chakra-button css-1k6w77y"
              type="button"
            >
              Programme
            </button>
            <button
              class="chakra-button css-1k6w77y"
              type="button"
            >
              Resources
            </button>
            <button
              class="chakra-button css-1k6w77y"
              type="button"
            >
              Blog
            </button>
            <button
              class="chakra-button css-1k6w77y"
              type="button"
            >
              Contact
            </button>
          </div>
        </div>
      </div>
    </nav>
  </div>
</div>
`;
